v4-client-js/src/clients/modules/post.ts

```diff
@@ -78,10 +81,20 @@ export class Post {
       messaging: () => Promise<EncodeObject[]>,
       gasPrice: GasPrice = GAS_PRICE,
       memo?: string,
+      account?: () => Promise<Account>,
     ): Promise<StdFee> {
-      const msgs = await messaging();
-      const account = await this.get.getAccount(wallet.address!);
-      return this.simulateTransaction(wallet.pubKey!, account.sequence, msgs, gasPrice, memo);
+      const msgsPromise = messaging();
+      const accountPromise = account ? (await account()) : this.account(wallet.address!);
```

Comment:
```suggestion
      const msgsPromise = messaging();
      const accountPromise = account ? account() : this.account(wallet.address!);
```
Since we are using promise.all later, we do not need to await here, since the await will unravel the promise.